Tests/MongoSwiftTests/CommandMonitoringTests.swift

```diff
@@ -0,0 +1,432 @@
+@testable import MongoSwift
+import Nimble
+import XCTest
+
+let cmPath = "Tests/Specs/command-monitoring/tests"
+let center =  NotificationCenter.default
+
+// TODO: don't hardcode this
+let VERSION = "3.6"
+
+final class CommandMonitoringTests: XCTestCase {
+    static var allTests: [(String, (CommandMonitoringTests) -> () throws -> Void)] {
+        return [
+            ("testCommandMonitoring", testCommandMonitoring)
+        ]
+    }
+
+    override func setUp() {
+        self.continueAfterFailure = false
+    }
+
+    func testCommandMonitoring() throws {
+        let client = try MongoClient()
+        try client.enableNotifications()
+        let testFiles = try FileManager.default.contentsOfDirectory(atPath: cmPath).filter { $0.hasSuffix(".json") }
+        for filename in testFiles {
+            // read in the file data and parse into a struct
+            let name = filename.components(separatedBy: ".")[0]
+
+            // remove this if/when bulkwrite is supported
+            if name.lowercased().contains("bulkwrite") { continue }
+
+            let testFilePath = URL(fileURLWithPath: "\(cmPath)/\(filename)")
+            let asDocument = try Document(fromJSONFile: testFilePath)
+            let testFile = try CMTestFile(fromDocument: asDocument)
+
+            print("-----------------------")
+            print("Executing tests for file \(name)...\n")
+
+            // execute the tests for this file
+            for var test in testFile.tests {
+
+                // Check that the current version we're testing is within bounds for this cases
+                if let maxVersion = test.maxServerVersion, maxVersion < VERSION {
+                    continue
+                } else if let minVersion = test.minServerVersion, VERSION < minVersion {
+                    continue
+                }
+
+                print("Test case: \(test.description)")
+
+                // 1. Setup the specified DB and collection with provided data
+                let db = try client.db(testFile.databaseName)
+                try db.drop() // In case last test run failed, drop to clear out DB
+                let collection = try db.createCollection(testFile.collectionName)
+                _ = try collection.insertMany(testFile.data)
+
+                // 2. Add an observer that looks for all events
+                var expectedEvents = test.expectations
+                let observer = center.addObserver(forName: nil, object: nil, queue: nil) { (notif) in
```

Comment:
FYI, `forName` could be `.commandSucceeded`, `.commandStarted`, etc. if you only want particular named events